test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -0,0 +1,272 @@
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.returns.ReturnsService
+
+import scala.concurrent.Future
+
+class InitialGainOrLossControllerSpec
+    extends ControllerSpec
+    with AuthSupport
+    with SessionSupport
+    with ReturnsServiceSupport
+    with ScalaCheckDrivenPropertyChecks
+    with RedirectToStartBehaviour {
+
+  implicit lazy val messagesApi: MessagesApi = controller.messagesApi
+
+  override val overrideBindings = List[GuiceableModule](
+    bind[AuthConnector].toInstance(mockAuthConnector),
+    bind[SessionStore].toInstance(mockSessionStore),
+    bind[ReturnsService].toInstance(mockReturnsService)
+  )
+
+  lazy val controller                  = instanceOf[InitialGainOrLossController]
+  implicit lazy val messages: Messages = MessagesImpl(Lang("en"), messagesApi)
+  val initialGainOrLosses              = sample[AmountInPence].copy(value = 2)
+
+  def sessionWithState(
+    answers: CompleteInitialGainOrLossAnswers,
+    disposalDate: DisposalDate
+  ): (SessionData, FillingOutReturn) = {
+    val journey = sample[FillingOutReturn].copy(
+      draftReturn = sample[DraftReturn].copy(
+        initialGainOrLossAnswers = Some(answers)
+      )
+    )
+
+    SessionData.empty.copy(journeyStatus = Some(journey)) -> journey
+  }
+
+  val allQuestionsAnswered: CompleteInitialGainOrLossAnswers = CompleteInitialGainOrLossAnswers(
+    initialGainOrLosses
+  )
+
+  val (session, journey) = sessionWithState(allQuestionsAnswered, sample[DisposalDate])
+  val updatedDraftReturn =
+    journey.draftReturn.copy(initialGainOrLossAnswers = Some(CompleteInitialGainOrLossAnswers(AmountInPence(0))))
+  val updatedSession = session.copy(journeyStatus = Some(journey.copy(draftReturn = updatedDraftReturn)))
+
+  "InitialGainOrLossController" should {
+    "show display" when {
+      def performAction(): Future[Result] = controller.enterInitialGainOrLoss()(FakeRequest())
+      "display a page returning 200" in {
+
+        val fillingOutReturn = sample[FillingOutReturn]
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+          )
+        }
+
+        val result: Future[Result] = performAction()
+        status(result) shouldBe OK
+      }
+
+      "have proper contents" in {
+        val fillingOutReturn = sample[FillingOutReturn]
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+          )
+        }
+
+        val result: Future[Result] = performAction()
+
+        contentAsString(result) should include("Initial gain or loss")
+        contentAsString(result) should include("Did you make an initial gain or loss?")
+      }
+
+      "display the page" in {
```

Comment:
this almost looks the same as the other specs above - why do you have three of the same thing in this spec?

Reply:
Kindda, but I've wrapped all into checkPageIsDisplayed, which kindda does everything I need including status and content.